app/src/components/validators.js

```diff
@@ -169,10 +202,84 @@ const validators = {
         errors.push({ value: value['email'], message: `Invalid value for required field \`contacts[${index}].email\`.` });
       }
       if (!models.contact.phone2(value['phone2'])) {
-        errors.push({ value: value['phone2'], message: `Invalid value for required field \`contacts[${index}].phone2\`.` });
+        errors.push({ value: value['phone2'], message: `Invalid value for field \`contacts[${index}].phone2\`.` });
       }
     });
 
+    if (!models.contact.firstName(obj.covidContact['firstName'])) {
+      errors.push({ value: obj.covidContact['firstName'], message: 'Invalid value for required field `covidContact.firstName`.' });
+    }
+    if (!models.contact.lastName(obj.covidContact['lastName'])) {
+      errors.push({ value: obj.covidContact['lastName'], message: 'Invalid value for required field `covidContact.lastName`.' });
+    }
+    if (!models.contact.phone1(obj.covidContact['phone1'])) {
+      errors.push({ value: obj.covidContact['phone1'], message: 'Invalid value for required field `covidContact.phone1`.' });
+    }
+    if (!models.contact.email(obj.covidContact['email'])) {
+      errors.push({ value: obj.covidContact['email'], message: 'Invalid value for required field `covidContact.email`.' });
+    }
+    if (!models.contact.phone2(obj.covidContact['phone2'])) {
+      errors.push({ value: obj.covidContact['phone2'], message: 'Invalid value for field `covidContact.phone2`.' });
+    }
+
+    if (!models.location.startDate(obj.location['startDate'])) {
+      errors.push({ value: obj.location['startDate'], message: 'Invalid value for required field `location.startDate`.' });
+    }
+    if (!models.location.endDate(obj.location['endDate'])) {
+      errors.push({ value: obj.location['endDate'], message: 'Invalid value for required field `location.endDate`.' });
+    }
+    if (!models.location.accTents(obj.location['accTents'])) {
+      errors.push({ value: obj.location['accTents'], message: 'Invalid value for required field `location.accTents`.' });
+    } else {
+      if (obj.location['accTents']) {
+        if (!validatorUtils.isNonEmptyString(obj.location['tentDetails'])) {
```

Comment:
Might be an open question of whether the details for the tents are required.
If we're requiring it here then I'll force it to be entered on the FE. Unless I'm reading this all bleary-eyed and this isn't a nonempty enforcement.